### PR TITLE
Sync Pane

### LINKED ISSUE #46: [Syncing] Pane
## Points to note
- Component in action - `Pane.js`
- Github Urls to this component:
  * [neeto-ui/Pane.js](https://github.com/bigbinary/neeto-ui/tree/67ee4a5188c5feccbeca2767107651fb24932764/lib/components/Pane.js)
  * [NitroUI/Pane.js](https://github.com/bigbinary/neeto-help-web/tree/0b11a7b0f98ac7c533df1e9ec37cf900b4cbc901/app/javascript/Admin/Components/NitroUI/Pane.js)
- :green_circle: Green color signifies how that particular line is written in `neeto-ui` package
- :red_circle: Red color signifies how that particular line is written in `NitroUI` internals

## Sync Required or Not
- Are there significant changes for `Pane.js` that needs to be synced to `neeto-ui` package from `NitroUI`? - **No**

## Diff
```diff
--- NitroUI/Pane.js	2020-11-28 16:54:24.011501400 +0530
+++ neetoui-official/Pane.js	2020-11-28 16:49:18.750476600 +0530
@@ -16,21 +16,21 @@
   } = props;
   return (
     <Drawer
-      className={classnames(["nh-pane--wrapper", className])}
+      className={classnames(["nui-pane--wrapper", className])}
       size={"40%"}
       canEscapeKeyClose
       canOutsideClickClose
       {...otherProps}
     >
       <div
-        className={classnames(["nh-pane--body"], {
-          "nh-pane--body__has-footer": showFooter
+        className={classnames(["nui-pane--body"], {
+          "nui-pane--body__has-footer": showFooter
         })}
       >
         {loading ? <PageLoader /> : children}
       </div>
       {showFooter && (
-        <div className="nh-pane--footer">
+        <div className="nui-pane--footer">
           <Button
             label="Cancel"
             size="large"
```

### PR DESCRIPTION
**Pull request by yedhink**
Fixes #46 

@edwinbbu _a

Please review and merge if everything looks good.

@goutham-subramanyam 

Had few questions. Why exactly are we naming this `nui-pane__footer`? I mean why mix it with `underscore` and `hyphen`? Why not just use `hyphen` alone? What does `__` signify?

### CODE REVIEW

**Comment by edwinbbu** [comment]
@goutham-subramanyam _a please review. There are changes in classes.

File: lib/styles/components/pane.scss [hunk 1]
```diff
@@ -1,4 +1,4 @@
-.bp3-drawer.nui-pane--wrapper {
+.bp3-drawer.nh-pane--wrapper {
```
Comment (edwinbbu) [inline lib/styles/components/pane.scss:1]: @yedhink _a name should be `nui` instead of `nh`

Comment (yedhink) [reply lib/styles/components/pane.scss:1]: Good catch. :+1:

**Comment by yedhink** [comment]
@edwinbbu _a

I have fixed that minor error.

I think this is good to be merged.

**Comment by edwinbbu** [comment]
@goutham-subramanyam _a Please review and merge.

**Comment by goutham-subramanyam** [comment]
@yedhink _a Please refer [this](https://github.com/bigbinary/neeto-help-web/pull/8766/files#diff-d34e9a819b35a62ad3a7cffbc32d0cbf8124f255781c7c0bbde682dbae970fcb) and update the PR.

-------

> Had few questions. Why exactly are we naming this nui-pane__footer? I mean why mix it with underscore and hyphen? Why not just use hyphen alone? What does __ signify?

Refer [BEM](http://getbem.com/introduction/)

**Comment by yedhink** [comment]
@goutham-subramanyam _a

I have updated the PR. Please review.

File: lib/components/Pane.js [hunk 2]
```diff
@@ -39,11 +39,16 @@ const Pane = props => {
             {...cancelButtonProps}
             dataTestId="cancel-button"
```
Comment (goutham-subramanyam) [inline lib/components/Pane.js:40]:
```suggestion
            data-test-id="cancel-button"
```

File: lib/components/Pane.js [hunk 3]
```diff
@@ -39,11 +39,16 @@ const Pane = props => {
...
           />
-          <Button label="Update" size="large" {...submitButtonProps} dataTestId="submit-button"/>
+          <Button
+            label="Update"
+            size="large"
+            {...submitButtonProps}
+            dataTestId="submit-button"
```
Comment (goutham-subramanyam) [inline lib/components/Pane.js:46]:
```suggestion
            data-test-id="submit-button"
```

File: src/index.js [hunk 4]
```diff
@@ -114,6 +123,21 @@ const App = () => {
           Settings
         </Tab.Item>
       </Tab>
+
+      <Button
+        dataTestId="open-pane"
```
Comment (goutham-subramanyam) [inline src/index.js:128]:
```suggestion
        data-test-id="open-pane"
```

File: lib/components/Pane.js [hunk 5]
```diff
@@ -16,30 +16,35 @@ const Pane = props => {
...
             className="mr-2"
             {...cancelButtonProps}
-            dataTestId="cancel-button"
+            data-test-id="cancel-button"
+          />
+          <Button
+            label="Update"
+            size="large"
+            {...submitButtonProps}
+            data-test-id="submit-button"
```
Comment (goutham-subramanyam) [inline lib/components/Pane.js:46]:
```suggestion
            data-test-id="submit-button"
            {...submitButtonProps}
```

File: lib/components/Pane.js [hunk 6]
```diff
@@ -16,30 +16,35 @@ const Pane = props => {
...
-        <div className="nui-pane--footer">
+        <div className="nui-pane__footer">
           <Button
             label="Cancel"
             size="large"
             style="secondary"
             className="mr-2"
             {...cancelButtonProps}
-            dataTestId="cancel-button"
+            data-test-id="cancel-button"
```
Comment (goutham-subramanyam) [inline lib/components/Pane.js:40]:
```suggestion
            data-test-id="cancel-button"
            {...cancelButtonProps}
```

File: lib/components/Pane.js [hunk 7]
```diff
@@ -16,30 +16,35 @@ const Pane = props => {
...
       </div>
       {showFooter && (
-        <div className="nui-pane--footer">
+        <div className="nui-pane__footer">
           <Button
             label="Cancel"
             size="large"
             style="secondary"
             className="mr-2"
             {...cancelButtonProps}
```
Comment (goutham-subramanyam) [inline lib/components/Pane.js:39]:
```suggestion
```

**Review by goutham-subramanyam — APPROVED** [review]
LGTM

**Pull request merged**